browser/ai_chat/ai_chat_brave_search_throttle_browsertest.cc

```diff
@@ -0,0 +1,239 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/ai_chat/content/browser/ai_chat_brave_search_throttle.h"
+
+#include <memory>
+#include <string>
+
+#include "base/files/file_path.h"
+#include "base/location.h"
+#include "base/path_service.h"
+#include "base/strings/string_util.h"
+#include "brave/browser/ui/brave_browser.h"
+#include "brave/browser/ui/sidebar/sidebar_controller.h"
+#include "brave/browser/ui/sidebar/sidebar_model.h"
+#include "brave/components/constants/brave_paths.h"
+#include "chrome/browser/ui/browser.h"
+#include "chrome/test/base/in_process_browser_test.h"
+#include "components/permissions/permission_request_manager.h"
+#include "components/permissions/test/mock_permission_prompt_factory.h"
+#include "content/public/test/browser_test.h"
+#include "content/public/test/browser_test_utils.h"
+#include "content/public/test/content_mock_cert_verifier.h"
+#include "content/public/test/test_navigation_observer.h"
+#include "content/public/test/test_utils.h"
+#include "net/base/net_errors.h"
+#include "net/dns/mock_host_resolver.h"
+#include "net/test/embedded_test_server/embedded_test_server.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace {
+
+constexpr char kBraveSearchHost[] = "search.brave.com";
+constexpr char kLeoPath[] = "/leo";
+constexpr char kOpenLeoButtonValidPath[] = "/open_leo_button_valid.html";
+constexpr char kOpenLeoButtonInvalidPath[] = "/open_leo_button_invalid.html";
+
+}  // namespace
+
+class AIChatBraveSearchThrottleBrowserTest : public InProcessBrowserTest {
```

Comment:
should be `PlatformBrowserTest`

Reply:
well, I guess some of this is necessarily desktop specific so maybe split it up as discussed in DM

Reply:
since there is no android implementation in this PR it can be done in a follow-up

Reply:
maybe add a TODO with a link to the issue to add android support?

Reply:
Issue created: https://github.com/brave/brave-browser/issues/41905
Comments added in https://github.com/brave/brave-core/pull/26082/commits/6cacbb06b4c7421bce8287978c0cd206d069dc0c